Sources/MongoSwift/BSON/BSONError.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+
+/// An empty protocol for encapsulating all errors that BSON package can throw.
+public protocol BSONErrorProtocol: LocalizedError {}
+
+/// A protocol describing errors caused by improper usage of the BSON library by the user.
+public protocol BSONUserError: BSONErrorProtocol {}
```

Comment:
Given that we only have 3 total error types, it might be better to just leave the "sub" protocols out for now. As we add more errors we can introduce a hierarchy to organize them.

Reply:
fine with me